util/chplenv/chpl_gmp.py

```diff
@@ -36,6 +36,7 @@ def get():
         else:
             gmp_val = 'none'
 
+    check_valid_var('CHPL_GMP', gmp_val, ['none', 'bundled', 'system'])
```

Comment:
Huh, this checks it even if it was the default. You could do that everywhere but if not IMO it's better to put this up top with the conditional
```
if gmp_val:
  check_valid_var('CHPL_GMP', gmp_val, ['none', 'bundled', 'system'])
else;
  ... set default ...
```

Let's pick one strategy and try to use it throughout.

Reply:
I prefer checking only if we don't try to infer something. I will adjust accordingly

Reply:
I take that comment back. When writing this I used always checking more consistency and for good reason. Many of these scripts already have complex logic, this would make it worse. I am going make them all always check the valid values (except for something like checking CHPL_GASNET_SEGMENT only with gasnet)